### PR TITLE
vmware_guest: assorted fixes and improvements

### PR DESCRIPTION
**Pull request by dagwieers**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

vmware_guest

##### ANSIBLE VERSION

v2.3

##### SUMMARY

A small collection of fixes and improvements:
- Simplify should_deploy_from_template()
- Bugfix for x.config that can be None (on corrupted VMs)
- Bugfix for mandatory guest_id (not when using templates)
- Simplify key testing and defaults
- Fix an incorrect reference to the last network
- Duplicate alias 'folder' removed

### CODE REVIEW

**Comment by dagwieers** [comment]
@nerzhul Please test this one too !

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 1]
```diff
@@ -606,8 +607,7 @@ def getvm(self, name=None, uuid=None, folder=None, name_match=None, cache=False)
                 paths = [x for x in paths if x.endswith(self.params['folder'])]
                 if len(paths) > 1:
                     self.module.fail_json(
-                        msg='%s matches more than one folder. Please use the absolute path starting with /vm/' %
-                            self.params['folder'])
+                        msg='%(folder)s matches more than one folder. Please use the absolute path starting with /vm/' % self.params)
```
Comment (nerzhul) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py:610]: please keep line length pep 8 compliant

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 2]
```diff
@@ -786,14 +786,17 @@ def remove_vm(self, vm):
             return {'changed': True, 'failed': False}
 
     def configure_guestid(self, vm_obj, vm_creation=False):
+        # guest_id is not required when using templates
+        if self.should_deploy_from_template():
```
Comment (nerzhul) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]: sorry, but if i have a template debian 8 and debian 9 is released, maybe i prefer to create debian 8 from debian 9 and change guestid. It's not the good solution. I think it's better to check the guestId value

Comment (nerzhul) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: Please change this check to if self.params['guest_id'] is None and self.should_deploy_from_template()

**Review by nerzhul — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_guest.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py:610]: Well, de PEP8 line-length was disabled from the checks. 80 columns by today's standards is no longer relevant and IIRC discussions wrt. this were not as strict. Besides the file is full of lines over the 80 column limit, I don't think this is a real concern. is it ?

    [dag@moria modules]$ awk '{ print length }' cloud/vmware/vmware_guest.py | sort | uniq -c | sort -n -k2 | tail
      4 114
      4 115
      2 116
      3 117
      1 118
      1 119
      1 120
      2 133
      1 142
      1 172

File: lib/ansible/modules/cloud/vmware/vmware_guest.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: Ok. As long as it is not mandatory, I guess. Fixed !

**Comment by nerzhul** [comment]
okay for me, shipit

**Review by nerzhul — APPROVED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_guest.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py:610]: I consider folding to 80 columns a bug

Comment (jtyr) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py:610]: IMHO everybody should aim for PEP8 compliance unless it has some very negative impact like creating new variables only to be able to fit into the 80 chars limit.

Comment (nerzhul) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py:610]: pycharm reports 120 chars not 80

**Pull request merged**